src/Controller/UserController.php

```diff
@@ -80,23 +81,57 @@ public function register(UserDto $userDto): Response
         return new JsonResponse($savedUserDto, Response::HTTP_CREATED);
     }
 
-//    /**
-//     * @Route(path="/{id}" methods={"DELETE"})
-//     */
-//    public function delete($id): Response
-//    {
-//        $user = $this->userRepository->findUserById($id);
-//
-//        if ($user) {
-//            $this->userRepository->remove($user);
-//            $this->logger->info('An user was deleted and removed from DB');
-//            return new JsonResponse();
-//        }
-//
-//
-//
-//        return new JsonResponse(['User with id' . $id . ' was deleted.'], Response::HTTP_OK);
-//    }
+    /**
+     * @Route(path="/{id}", methods={"DELETE"})
+     */
+    public function deleteUserAction($id): Response
+    {
+        $user = $this->userRepository->findUserById($id);
+
+        if (null === $user) {
+            return new JsonResponse('No user found', Response::HTTP_BAD_REQUEST);
+        }
+        $this->userRepository->remove($user);
+        $this->userRepository->add($user);
+        $this->logger->info('An user was soft-deleted');
+
+        return new JsonResponse(['User with id' . $id . ' was soft-deleted.'], Response::HTTP_OK);
+    }
+
+    /**
+     * @Route(path="/recover", methods={"POST"})
+     */
+    public function unDeleteUserAction(Request $request): Response
+    {
+        $data = $request->getContent();
+        $decodedData = json_decode($data, true);
+        if (!isset($decodedData['email'])) {
+            $this->logger->warning('An atempt to post on recover without email');
+            return new JsonResponse(
+                'You should provide an email',
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+        $this->entityManager->getFilters()->disable('softdeleteable');
+        $user = $this->userRepository->findOneBy(['email' => $decodedData['email']]);
+        if (null === $user) {
+            $this->logger->warning('No user for recover', ['email' => $decodedData['email']]);
+            return new JsonResponse(
+                'No user with provided email',
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+        $user->setDeletedAt(null);
+        $this->userRepository->add($user);
```

Comment:
it's very weird that your "update" method on a user is called `add`.. it looks like you create a new user here

Reply:
Yes. Same as first observation... a lot of inconsistencies

Reply:
Solved.